pkg/rtc/participant.go

```diff
@@ -755,16 +718,7 @@ func (p *ParticipantImpl) MaybeStartMigration(force bool, onStart func()) bool {
 		// DownTrack close has checks to handle the case of closing before bind.
 		// So, an `Unbind` before close would bypass that logic.
 		//
-		p.lock.Lock()
-		downTracksToClose := make([]*sfu.DownTrack, 0, len(p.subscribedTracks))
-		for _, st := range p.subscribedTracks {
-			downTracksToClose = append(downTracksToClose, st.DownTrack())
-		}
-		p.lock.Unlock()
-
-		for _, dt := range downTracksToClose {
-			dt.CloseWithFlush(false)
-		}
+		p.SubscriptionManager.Close(true)
```

Comment:
This is going to cause problems I think on migration. This one is closing tracks synchronously. That is required before closing the subscriber peer connection because of the reason mentioned in the comments ☝️ . But, subscription manager is closing tracks in a `goroutine`.